pkg/cri/server/podsandbox/sandbox_stop.go

```diff
@@ -83,24 +88,25 @@ func (c *Controller) stopSandboxContainer(ctx context.Context, sandbox sandboxst
 			if !errdefs.IsNotFound(err) {
 				return fmt.Errorf("failed to wait for task: %w", err)
 			}
-			return cleanupUnknownSandbox(ctx, id, sandbox)
+			return cleanupUnknownSandbox(ctx, id, podSandbox)
 		}
 
 		exitCtx, exitCancel := context.WithCancel(context.Background())
 		stopCh := make(chan struct{})
 		go func() {
 			defer close(stopCh)
-			exitStatus, exitedAt, err := c.waitSandboxExit(exitCtx, id, exitCh)
+			exitStatus, exitedAt, err := c.waitSandboxExit(exitCtx, podSandbox, exitCh)
 			if err != nil && err != context.Canceled && err != context.DeadlineExceeded {
 				e := &eventtypes.SandboxExit{
 					SandboxID:  id,
 					ExitStatus: exitStatus,
 					ExitedAt:   protobuf.ToTimestamp(exitedAt),
 				}
-				log.G(ctx).WithError(err).Errorf("Failed to wait sandbox exit %+v", e)
+				log.G(ctx).WithError(err).Errorf("Failed to wait pod sandbox exit %+v", e)
 				// TODO: how to backoff
 				c.cri.BackOffEvent(id, e)
 			}
+			podSandbox.Exit(*containerd.NewExitStatus(exitStatus, exitedAt, err))
```

Comment:
I don't think it's correct. Line 107 has backoff event. And then the code still goes into set `Exit`.
And the `Exit` has close the channel. I think it might cause leak issue.

Reply:
I  think we can remove the codes of backoff event here because the `startSandboxExitMonitor` is responsible to handle backoff when sandbox exit event handle failed. So I removed the backoff logic here.

Reply:
changed to call podSandbox.Exit when the error is not a context error.